src/decisionengine/framework/taskmanager/TaskManager.py

```diff
@@ -66,11 +66,13 @@ def _create_module_instance(config_dict, base_class, channel_name):
             class_name = _find_only_one_subclass(my_module, base_class)
 
     class_type = getattr(my_module, class_name)
-    #here we add the channel name to the LogicEngine class to get the correct logger
-    if class_name == "LogicEngine":
-        return class_type(config_dict["parameters"], channel_name)
-    else:
-        return class_type(config_dict["parameters"])
+
+    param_config_dict = config_dict["parameters"].copy()
+
+    if "channel_name" not in param_config_dict.keys():
+        param_config_dict["channel_name"] = channel_name        
+
+    return class_type(param_config_dict)
```

Comment:
Could this be simplified down to something like:

```python
if "channel_name" not in config_dict["parameters"].keys():
    return  config_dict.get("name")
else:
   return getattr(my_module, class_name)
```

Reply:
That makes more sense.  Mostly I was just noticing the clone, store, use once and discard.

Reply:
If the configuration has channel_name should take precedence. This could work if all `config_dict["parameters"]` keys are guaranteed to be strings:
```python
return class_type(dict(channel_name=channel_name, **config_dict["parameters"]))
```
This is safer in case a key is not a string:
```python
if "channel_name" in config_dict["parameters"]:
    return  class_type(config_dict["parameters"])
else
    return class_type(dict(config_dict["parameters"], channel_name=channel_name))
```